base/io.jl

```diff
@@ -902,7 +902,7 @@ end
 function readuntil(io::IO, target::AbstractVector{T}; keep::Bool=false) where T
     out = (T === UInt8 ? StringVector(0) : Vector{T}())
     readuntil_vector!(io, target, keep, out)
-    return out
+    return out::Vector{T}
```

Comment:
It seems to me that if we know `T` we can already infer this, and if we don't know `T` there is no difference?

Reply:
That sounds right. This might have been something I added hoping it would help, but then found the real answer later. I am checking now whether omitting this makes any difference, though I am strongly inclined to agree with you.

I need to get in the habit of specifying which packages I happened to be studying for invalidations when I make individual commites. Otherwise I struggle to re-create the test case that led to this. We should really think about #36393 a bit.

Reply:
Seems fine without (checked Images, Plots, FilePathsBase, and of course indirectly all the packages they depend on).